pytket/stub_generation/regenerate_stubs

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+import os
+import re
+import subprocess
+from pathlib import Path
+import argparse
+
+
+def replace_in_file_string(file_string: str, matcher: str, replacement: str) -> tuple[str, int]:
+    split_text = re.split(matcher, file_string)
+    modified_text = replacement.join(split_text)
+    return modified_text, len(split_text) - 1
+
+
+def replace_in_file_string_multiline(file_string: str, matcher: str, replacement: str) -> tuple[str, int]:
+    split_text = re.split(matcher, file_string, re.MULTILINE)
+    modified_text = replacement.join(split_text)
+    return modified_text, len(split_text) - 1
+
+
+def handle_args_kwargs_types(file_string: str) -> tuple[str, int]:
+    modified, n_args = replace_in_file_string(file_string, "\*args", "*args: Any")
+    modified, n_kwargs = replace_in_file_string(modified, "\*\*kwargs", "**kwargs: Any")
+    n_total = n_args + n_kwargs
+    if n_total > 0 and not re.search("from typing import .*Any", modified):
+        modified = "from typing import Any" + os.linesep + modified
+    return modified, n_total > 0
+
+
+def handle_numpy_stuff(file_string: str) -> tuple[str, bool]:
+    split_text = re.split("numpy\.ndarray\[", file_string)
+    modified_text = "NDArray[".join(split_text)
+    if len(split_text) > 1:
+        modified_text, _ = replace_in_file_string(modified_text, "\[.,.\]", "")
+        modified_text, _ = replace_in_file_string(modified_text, "NDArray\[bool\]", "NDArray[numpy.bool_]")
+        modified_text = "from numpy.typing import NDArray" + os.linesep + modified_text
+        return modified_text, True
+    return file_string, False
+
+
+def handle_circuit_depth_problem(file_string: str) -> str:
+    split_text = re.split(f'@overload{os.linesep}\s+def depth\(self\) -> int: \.\.\.', file_string, re.MULTILINE)
+    modified = "def depth(self) -> int: ...".join(split_text)
+    split_text = re.split(f'@overload{os.linesep}\s+def depth\(self\) -> Any: \.\.\.{os.linesep}\s+', modified, re.MULTILINE)
+    modified = "".join(split_text)
+    split_text = re.split(f'@overload{os.linesep}\s+def depth_2q\(self\) -> int: \.\.\.', modified, re.MULTILINE)
+    modified = "def depth_2q(self) -> int: ...".join(split_text)
+    split_text = re.split(f'@overload{os.linesep}\s+def depth_2q\(self\) -> Any: \.\.\.{os.linesep}\s+', modified, re.MULTILINE)
+    modified = "".join(split_text)
+    return modified
+
+
+if __name__ == "__main__":
```

Comment:
Ah yes. Its because afterwards I ran `black` on the files to fix the linting CI, which changed the format. Honestly I preferred the format without black. Is there a good way to exclude them from black? Or would you prefer to have them formatted by black?

And yes your CI suggestion sounds good!

Reply:
Since they're autogenerated I'm fine with not formatting them with black. I'd hope there should be a way to configure black to ignore `.pyi` files?